### PR TITLE
modernize cmake

### LINKED ISSUE #342: cleanup cmake and adhere to modern cmake

### PR DESCRIPTION
**Pull request by XzzX**
closes #342

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -80,9 +77,20 @@ endif()
 get_target_property(ESPP_CFLAGS _espressopp COMPILE_OPTIONS)
 message(STATUS "_espressopp compiler flags are: ${ESPP_CFLAGS}")
 
-target_link_libraries(_espressopp ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} MPI::MPI_CXX ${FFTW3_LIBRARIES} ${VAMPIRTRACE_LIBRARIES})
+target_include_directories(_espressopp PUBLIC ${CMAKE_SOURCE_DIR}/src)
```
Comment (junghans) [inline src/CMakeLists.txt:80]: All these 3 should be marked e.g. `$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/src>` as they aren't useful after install.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -80,9 +77,20 @@ endif()
...
+target_include_directories(_espressopp PUBLIC ${CMAKE_SOURCE_DIR}/src/include)
+target_include_directories(_espressopp PUBLIC ${CMAKE_BINARY_DIR}/src)
+target_include_directories(_espressopp PRIVATE ${FFTW3_INCLUDES})
```
Comment (junghans) [inline src/CMakeLists.txt:83]: We should create an imported target in our custom `FindFFTW3.cmake`.

See  https://github.com/votca/tools/blob/master/CMakeModules/FindFFTW3.cmake#L40-L51

File: CMakeLists.txt [hunk 3]
```diff
@@ -56,16 +56,13 @@ find_package(MPI REQUIRED COMPONENTS CXX)
...
 
 find_package(FFTW3 REQUIRED)
-include_directories(${FFTW3_INCLUDES})
 
 ########################################################################
 #Process GROMACS settings
 ########################################################################
 
 if(WITH_XTC)
     find_package(GROMACS REQUIRED)
```
Comment (junghans) [inline CMakeLists.txt:65]: We should install this library in our CI and delete our custom find module as gromacs brings it's own now.

Comment (junghans) [reply CMakeLists.txt:65]: I.e. `find_package(GROMACS 2021 CONFIG NAMES gromacs gromacs_d)`

File: src/CMakeLists.txt (hunk 2 above)
Comment (XzzX) [reply src/CMakeLists.txt:83]: Also true for `mpi4py`. Anyone volunteering? :)

File: CMakeLists.txt (hunk 3 above)
Comment (XzzX) [reply CMakeLists.txt:65]: Yes.

General remark. As far as I can see Gromacs is only used to write `XTC` files. Perhaps it is a little bit of an overkill to require a Gromacs installation just for that...

Comment (junghans) [reply CMakeLists.txt]: It is optional after all.

Comment (XzzX) [reply CMakeLists.txt]: True. Is Gromacs installed in our buildenv?

File: src/CMakeLists.txt (hunk 2 above)
Comment (junghans) [reply src/CMakeLists.txt]: I would leave that until #351 is done.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -80,9 +58,19 @@ endif()
 get_target_property(ESPP_CFLAGS _espressopp COMPILE_OPTIONS)
 message(STATUS "_espressopp compiler flags are: ${ESPP_CFLAGS}")
 
-target_link_libraries(_espressopp ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} MPI::MPI_CXX ${FFTW3_LIBRARIES} ${VAMPIRTRACE_LIBRARIES})
+target_include_directories(_espressopp PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/src> $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/src/include>
+     $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/src>)
```
Comment (XzzX) [inline src/CMakeLists.txt:62]: Do we also need `INSTALL_INTERFACE` for the installed version to work?

File: contrib/mpi4py/CMakeLists.txt [hunk 5]
```diff
@@ -4,14 +4,14 @@ set(MPI4PY_DIR mpi4py-3.0.3)
 set(MPI4PY_SOURCES ${MPI4PY_DIR}/src/MPI.c)
 set(DL_SOURCES ${MPI4PY_DIR}/src/dynload.c)
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/${MPI4PY_DIR}/src)
-
 add_library(MPI ${MPI4PY_SOURCES})
-target_link_libraries(MPI ${Python3_LIBRARIES} ${VAMPIRTRACE_LIBRARIES} MPI::MPI_CXX)
+target_link_libraries(MPI PUBLIC Python3::Python MPI::MPI_CXX)
+target_include_directories(MPI PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/${MPI4PY_DIR}/src)
```
Comment (XzzX) [inline contrib/mpi4py/CMakeLists.txt:9]: Do we need `BUILD_INTERFACE` here as well?

**Review by XzzX — COMMENTED** [review]
Thank you @junghans. We are getting somewhere :)

File: src/CMakeLists.txt (hunk 4 above)
Comment (junghans) [reply src/CMakeLists.txt:62]: We don't install any headers, so there is no install interface ;-)

File: contrib/mpi4py/CMakeLists.txt (hunk 5 above)
Comment (junghans) [reply contrib/mpi4py/CMakeLists.txt:9]: Good catch!

File: src/CMakeLists.txt (hunk 4 above)
Comment (XzzX) [reply src/CMakeLists.txt:62]: Ok, so we do not use the raw usage of `_espresso.so` without the Python layer? @stuehn mentioned it once. We do not have raw C++ users?

**Comment by junghans** [comment]
I think this ready.

**Comment by XzzX** [comment]
Fine from my side.

File: src/CMakeLists.txt (hunk 4 above)
Comment (junghans) [reply src/CMakeLists.txt:62]: We can always add an install interface once we have users for it.

**Pull request merged**